specs/agents/tracing-instrumentation-aws.md

```diff
@@ -0,0 +1,53 @@
+## AWS services spans
+
+We describe how to instrument some of AWS' services in this document.
+Many of the services have a fairly small API beyond the basic administrative commands. We therefore
+only instrument the most interesting and often-repeated operations.
+
+### S3 (Simple Storage Service)
+
+AWS Simple Storage Service offers object storage via a REST API. The objects are organized into buckets, which are 
+themselves organized into regions.
+
+- `span.name`: The span name should follow this pattern: `S3 <OPERATION_NAME> <BUCKET-NAME>`. For example,
+`S3 GET_OBJECT myBucket`. Note that the operation name is in all caps with underscores (snake case in caps).
+- `span.type`: `storage`
+- `span.subtype`: `s3`
+- `span.action`: The operation name in snake case in lower case. For example ‘get_object’. 
+
+#### Span context fields
+
+- **`context.destination.address`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.port`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.region`**: mandatory. The AWS region where the bucket is.
+- **`context.destination.service.name`**: mandatory. Use `s3`
+- **`context.destination.service.resource`**: mandatory. The bucket ARN, if available, otherwise simply `s3/<BUCKET-NAME`.
+Note that the ARN will contain the bucket name. The s3 API allows either the bucket name or the ARN to be provided when
+referring to a bucket so use whichever the user has provided.
+- **`context.destination.service.type`**: mandatory. Use `storage`.
+
+### DynamoDB
+
+AWS DynamoDB is a document database so instrumenting it will follow the [db spec](tracing-instrumentation-db.md). 
+Note that the `db.instance` field is 
+the AWS region.
```

Comment:
Should we also capture the region in `context.destination.region` so that it's consistent with the other services?

Reply:
Maybe both places? Seems like `db.instance` is a field people expect to see for DB spans, but with the addition of `context.destination.region` we should definitely populate it as well.

Reply:
Yes, that's what I was thinking 👍🏼